bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/Workbench.java

```diff
@@ -2212,6 +2214,11 @@ public Object compute(IEclipseContext context, String contextKey) {
 			}
 		});
 		WorkbenchPlugin.getDefault().initializeContext(e4Context);
+
+		BundleContext context = FrameworkUtil.getBundle(Platform.class).getBundleContext();
```

Comment:
Please just use Declarative services here to register the instance, that is much more flexible and you dint need to get the bundle context.

Reply:
Can you please propose how to do so?

Reply:
See for example `org.eclipse.ui.internal.CommandToModelProcessor` or `org.eclipse.ui.internal.BindingToModelProcessor`

Reply:
Looking at given examples it **seem** that you want Workbench be annotated as `@Component(service = { IActivityManagerProxy.class }` but I'm not sure if that is what is needed here and which side effects that would have for such a central class.

Can we do following: let us first get this thing merged in known working state and after that you could refactor it to apply best services practices?

Reply:
> Looking at given examples it seem that you want Workbench be annotated as @Component(service = { IActivityManagerProxy.class } 

No just annotate `IActivityManager` implementation